src/gen_fn_interface.jl

```diff
@@ -272,10 +272,25 @@ that if the original `trace` was generated using non-default argument values,
 then for each optional argument that is omitted, the old value will be
 over-written by the default argument value in the updated trace.
 """
-function update(trace, args::Tuple, argdiffs::Tuple, ::ChoiceMap)
+function update(trace, args::Tuple, argdiffs::Tuple, constraints::ChoiceMap)
     error("Not implemented")
 end
 
+"""
+    update(trace;
+        args::Tuple=get_args(trace),
+        argdiffs::Tuple=map((_) -> NoChange(), args),
+        constraints::ChoiceMap=EmptyChoiceMap())
+
+Form of `update` with keyword arguments providing common defaults.
+"""
+function update(trace;
+        args::Tuple=get_args(trace),
+        argdiffs::Tuple=map((_) -> NoChange(), args),
```

Comment:
Should the default be `NoChange` or `UnknownChange`?

Reply:
Now there are two convenience forms:

If you provide args, then argdiffs is an optional keyword argument that defaults to UnknownChange. (This is so you don't need to construct the unknown change argdiffs).

If you do not provide args then argdiffs is automatically set to NoChanges. (This is a common case e.g. for top-level generative functions on which we are doing MCMC or optimization).